### PR TITLE
Add output-dir option to flow generate

### PR DESCRIPTION
**Pull request by tatiana**
Fix: #1529

### CODE REVIEW

**Comment by tatiana** [comment]
I'm adding this tiny change so that we don't block #1520

File: sql-cli/tests/test___main__.py [hunk 1]
```diff
@@ -245,6 +245,25 @@ def test_generate_invalid(workflow_name, message, initialised_project_with_tests
...
+            initialised_project.directory.as_posix(),
+            "--output-dir",
+            initialised_project.directory.as_posix(),
+        ],
+    )
+    expected_dag_file_path = initialised_project.directory / "example_basic_transform.py"
+    assert expected_dag_file_path.exists()
+    message = "has been successfully generated."
+    assert message in result.stdout
+    assert not result.exit_code
```
Comment (feluelle) [inline sql-cli/tests/test___main__.py:264]:
```suggestion
    assert result.exit_code == 0
```
I think this shows more clearly that you expect it to succeed.

File: sql-cli/tests/test___main__.py [hunk 2]
```diff
@@ -247,6 +247,25 @@ def test_generate_invalid(workflow_name, message, initialised_project_with_tests
     assert message in result.stdout
 
 
+def test_generate_with_output_dir(initialised_project):
```
Comment (pankajkoti) [inline sql-cli/tests/test___main__.py:250]: how about we pass the tmp_path as a fixture here and use that as the output_dir ?

**Review by pankajkoti — APPROVED** [review]
LGTM, minor suggestion in the test if you feel can be checked.

File: sql-cli/tests/test___main__.py (hunk 2 above)
Comment (tatiana) [reply sql-cli/tests/test___main__.py]: Great suggestion, @pankajkoti, particularly now that we are reusing the initialised project per test module!

**Review by feluelle — APPROVED** [review]

**Pull request merged**